### PR TITLE
FIX: Locale mismatch at theme translations picker

### PR DESCRIPTION
**Pull request by Grubba27**
Before, the theme translations picker value was set to the site's default locale, which mismatches from the user's locale.

This commit changes the picker value to the user locale.

relates to https://meta.discourse.org/t/locale-mismatch-at-theme-translations/302879

<details>
<summary>Before</summary>

Screenshot 2024-04-19 at 11 46 13

_You can see in the left-right corner that the picker is not consistent_

</details>
<details>
<summary>Now</summary>

Screenshot 2024-04-19 at 11 42 54

</details>

> [!NOTE]
> Is there a better way to get the user's locale session? I used ruby [I18n.locale](https://github.com/discourse/discourse/blob/a45b2b1124e71a1d9fd8af2b53f69efdf5dba2af/app/assets/javascripts/admin/addon/routes/admin-customize-themes-show.js#L44) for [consistency](https://stackoverflow.com/questions/673905/how-can-i-determine-a-users-locale-within-the-browser).

### CODE REVIEW

File: spec/system/admin_customize_themes_spec.rb
Comment (ZogStriP) [inline spec/system/admin_customize_themes_spec.rb:164]: Probably better to leave that link in the commit message rather than in the code.

File: app/controllers/admin/themes_controller.rb
Comment (pmusaraj) [inline app/controllers/admin/themes_controller.rb:179]: Maybe use `effective_locale` from the `User` model? Seems like the right thing, although `I18n.locale` seems to be doing the job too.

**Review by pmusaraj — APPROVED** [review]
Looks good (took it for a spin locally), thanks!

File: app/controllers/admin/themes_controller.rb
Comment (Grubba27) [reply app/controllers/admin/themes_controller.rb:179]: Seems more correct indeed.

I'll change it

File: spec/system/admin_customize_themes_spec.rb
Comment (Grubba27) [reply spec/system/admin_customize_themes_spec.rb:164]: Sure, the link is in the commit as well.

I'll remove the comment

**Comment by pmusaraj** [comment]
Cool! Once specs pass, go ahead and press that Merge button.

**Pull request merged**
